scenarios/bigquery.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python
+
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Runs a bigquery query, filters the results, and uploads the filtered and complete data to GCS."""
+
+import argparse
+import subprocess
+import sys
+import time
+import json
+
+DEFAULTBUCKET = "gs://k8s-metrics"
+DEFAULTPROJECT = "k8s-gubernator"
+
+def check(*cmd, **keyargs):
+    """Logs and runs the command, raising on errors."""
+    print >>sys.stderr, 'Run:', cmd
+    # If "stdin_string" keyword arg is present run command and communicate string to stdin
+    if "stdin_string" in keyargs:
```

Comment:
I tried something similar to that at first, but StringIO doesn't implement fileno() so it cannot be used as a file by subprocess.check_call. I changed the key "stdin_string" to "stdin" though for uniformity.